### ISSUE TITLE
export wallet tx history in the backup

### ISSUE THREAD

**Issue by stepansnigirev** [cryptoadvance/specter-desktop#472]
We could include tx history with merkle proofs in specter backup, then it will be very easy to move wallets from one Specter to another or to recover from the backup - we wouldn't need a full rescan, but only from the last seen block.

**Comment by decentralizedb** [comment]
Could be interesting to have the possibility to export the tx labels too.

**Comment by VWVideo** [comment]
Yes, been experimenting with this. I just exported my wallet backups on my main PC, and loaded them up on my spare. Everything came through perfectly, but all of the labels were gone. Having labels come thru in the restore is especially critical for people doing coin control and selection. I would be absolutely lost without my labels, so there needs to be a way to back them up.

**Comment by ChristopherA** [comment]
This is something that Blockchain has on its to do list, which is the ability to export your Account Book with all metadata associated with the wallet addresses associated with it. We’d like an cross-wallet format for this.

**Comment by al-munazzim** [comment]
**Triage Assessment**

**Verdict:** VALID-ENHANCEMENT / LOW priority

**Summary:** Request to include transaction history with merkle proofs in wallet backups, enabling fast recovery without full rescan.

**Analysis:**
- Currently Specter backups contain wallet descriptors and device info, but not tx history or labels
- Full rescan on restore is slow and painful, especially for large wallets
- BIP-329 (wallet labels export) has since been standardized and partially addresses the labels portion
- The merkle proof approach for tx history is technically sound but adds significant complexity
- Related: Blockchain Commons expressed interest in cross-wallet metadata format (comment from @ChristopherA)

**Recommendation:**
1. Labels: adopt BIP-329 export/import as a near-term improvement (lower complexity)
2. Tx history with proofs: defer — high implementation effort, niche benefit given that rescan works (just slowly)
3. Consider splitting into two issues: labels backup (BIP-329) and tx history backup (merkle proofs)

**Impact:** Low-medium. Improves UX for backup/restore workflow but has workarounds (rescan). Labels are the higher-value piece.

*This is an AI-generated triage assessment. Feedback welcome.*
